File: GenerateInitvector/GenerateRandomInitvector.php
<?php
function generateRandomInitvector()
{
    return openssl_random_pseudo_bytes(16, $crypto_strong);
}

function base64Encoding($input)
{
    return base64_encode($input);
}

echo 'Generate a 16 byte long Initialization vector (IV)' . PHP_EOL;
$iv = generateRandomInitvector();
$ivBase64 = base64Encoding($iv);
echo 'generated iv length: ' . strlen($iv) . ' data: '. $ivBase64 . PHP_EOL;
?>

File: GenerateInitvector/GenerateRandomInitvector.cs
using System;
using System.Security.Cryptography;

public class GenerateAesKey {
	public static void Main() {
		Console.WriteLine("Generate a 16 byte long Initialization vector (IV)");
    byte[] iv = GenerateRandomInitvector();
    string ivBase64 = Base64Encoding(iv);
    Console.WriteLine("generated iv length: " + iv.Length + " data: " + ivBase64);
	}

  static byte[] GenerateRandomInitvector() {
    RNGCryptoServiceProvider rngCsp = new RNGCryptoServiceProvider();
    byte[] iv = new byte[16];
    rngCsp.GetBytes(iv);
    return iv;
  }

  static string Base64Encoding(byte[] input) {
    return Convert.ToBase64String(input);
  }
}

